modules/earthengine-layers/src/earth-engine-layer.js

```diff
@@ -112,6 +115,12 @@ export default class EarthEngineLayer extends CompositeLayer {
       if (!eeObject.getFilmstripThumbURL) {
         throw new Error('eeObject must have a getFilmstripThumbURL method to animate.');
       }
+    } else if (props.asVector) {
+      renderMethod = 'vector';
+      const geojsonUrl = await promisifyEEMethod(eeObject, 'getDownloadURL', 'json', ['.geo'], '');
+      const resp = await fetch(geojsonUrl);
+      const geojsonData = await resp.json();
```

Comment:
I should probably use `JSONLoader` here instead of `fetch`.

Reply:
Agreed, streaming JSONLoader seems valuable given the potential size of these geojsons.

Reply:
I haven't gotten streaming to work yet. I was trying to use `loadInBatches` and the suggestion from the deck.gl docs [here](https://deck.gl/#/documentation/developer-guide/performance-optimization?section=handle-incremental-data-loading), but think I'm having trouble appending to the array held in `state`, because I can see the features loading one by one, but `geojsonDataChunks` in `state` is empty.

Reply:
Well you could still load it in streaming mode, but only display when fully received. That is gentler on the main thread. Then as step 2 we can display it in streaming mode as well as the batches come in.